app/src/main/java/org/mozilla/focus/utils/TLDExtractor.kt

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.utils
+
+import android.content.Context
+import kotlinx.coroutines.experimental.runBlocking
+import java.io.BufferedReader
+import java.io.IOException
+import java.io.InputStream
+import java.io.InputStreamReader
+import java.io.Reader
+import java.util.ArrayList
+import java.util.Collections
+import java.util.Comparator
+import java.util.regex.Pattern
+
+/**
+ * Finds the Top Level (or Secondary Level) Domain using a public list of TLD domains: http://publicsuffix.org/list/
+ * Initialize as soon as possible in application.
+ * Based on:
+ *  http://www.supermind.org/blog/1078/extracting-second-level-domains-and-top-level-domains-tld-from-a-url-in-java
```

Comment:
Note that we have such code in Fennec already:
https://searchfox.org/mozilla-central/source/mobile/android/base/java/org/mozilla/gecko/util/publicsuffix/PublicSuffix.java

The code in Fennec was based on Guava and uses a minimal list and can do lookups quite efficiently.